src/neuroconv/tools/nwb_helpers/_configure_backend.py

```diff
@@ -15,9 +16,39 @@ def configure_backend(
     """Configure all datasets specified in the `backend_configuration` with their appropriate DataIO and options."""
     nwbfile_objects = nwbfile.objects
 
+    # When a configuration is generated on an equivalent NWBFile elsewhere in memory, the object IDs will not match
+    # But a mapping can be formed to reconcile the two
+    # This is the case for the interface and converter usage of the feature (since they are separate workflow steps)
+    # As well as the NWB GUIDE (since they are separate endpoints in the Flask server)
```

Comment:
This block was the trickiest part of this, but a necessary one as explained

There was technically a second approach where each interface/converter is able to generate and maintain the in-memory `self.nwbfile` object to use across different steps of the workflow, but that could have all kinds of weird consequences and add more complexity to deal with inside each step, so I concluded the simplest approach was to just equivocate two configurations generated from equivalent file structures

Reply:
I suppose an alternative approach could be to define this logic in something like a `.update_object_ids` method on the `BackendConfiguration` class - what do you think @h-mayorquin?

Reply:
Moved to https://github.com/catalystneuro/neuroconv/pull/816